Categories bug fixed and tests added

kdefliese opened the pull request:
Fixed bug preventing categories page from loading
Added specs for new category page
Added cucumber tests for new and edit category actions

kdefliese committed:
deployed to heroku

kdefliese committed:
started writing feature for creating categories

kdefliese committed:
cucumber test seems to be failing appropriately

kdefliese committed:
updated feature based on better understanding of the bug

kdefliese committed:
added rspec tests for rendering index and new page

kdefliese committed:
updated category new_or_edit function to handle nil ID, rspecs passing

kdefliese committed:
cucumber test passes for going to the categories page

kdefliese committed:
added to cucumber test so it covers category creation too

kdefliese committed:
added cucumber feature for editing categories

kariabancroft commented:
Files in the DB folder should be deleted from PR and added to gitignore

kdefliese committed:
Added db files to gitignore

kariabancroft commented on features/create_category.feature:
```diff
@@ -0,0 +1,14 @@
+  Scenario: Successfully create new Categories
+    When I click on Categories
+    Then I should be on the new category page
+    When I fill in "category_name" with "Meow Meow"
```
What about the other fields available? Do you want to test any of those?

kdefliese committed:
deleted db files

kariabancroft commented on features/edit_category.feature:
```diff
@@ -0,0 +1,17 @@
+    And I am logged into the admin panel
+    And I have an existing Category named "Meow"
+
+  Scenario: Successfully edit existing Categories
```
Do you want to test any non-success cases?

kariabancroft commented on spec/controllers/admin/post_types_controller_spec.rb:
```diff
@@ -10,7 +10,7 @@
   end
 
-  it "index shoudld redirect to new" do
+  it "index should redirect to new" do
```
+1 for fixing existing code!

kdefliese committed:
added additional field validation for category add/edit

kdefliese committed:
added cucumber scenarios for saving with blank category name

kdefliese committed:
fixed error that was happening when saving category with empty name

kdefliese committed:
added last step for cucumber tests for saving nil category names

kdefliese commented:
Updated:
- removed db files and added to gitignore
- added lines to test scenarios for other category fields
- added cucumber scenarios for creating and editing categories with nil names
- fixed bug that was causing the page to show a rails error instead of a flash error message when doing the above

kdefliese committed:
removed db files again..

kariabancroft commented:
Looks good to close!
